app/Http/Controllers/DuesPackageController.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\DuesPackage;
+
+class DuesPackageController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $packages = DuesPackage::all();
+        return response()->json(['status' => 'success', 'dues_packages' => $packages]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'name' => 'required|string',
+            'eligible_for_shirt' => 'boolean',
+            'eligible_for_polo' => 'boolean',
+            'effective_start' => 'required|date',
+            'effective_end' => 'required|date',
+            'cost' => 'required|numeric'
+        ]);
+
+        try {
+            $package = DuesPackage::create($request->all());
+        } catch (QueryException $e) {
+            $errorMessage = $e->errorInfo[2];
+            return response()->json(['status' => 'error', 'message' => $errorMessage], 500);
+        }
+
+        if (is_numeric($package->id)) {
+            $dbPackage = DuesPackage::findOrFail($package->id);
+            return response()->json(['status' => 'success', 'dues_package' => $dbPackage], 201);
+        } else {
+            return response()->json(['status' => 'error', 'message' => 'Unknown error.'], 500);
+        }
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $package = DuesPackage::find($id);
+        if ($package) {
+            return response()->json(['status' => 'success', 'dues_package' => $package]);
+        } else {
+            return response()->json(['status' => 'error', 'message' => 'Payment not found.'], 404);
+        }
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $this->validate($request, [
+            'id' => 'required|exists:dues_packages',
+            'name' => 'string',
+            'eligible_for_shirt' => 'boolean',
+            'eligible_for_polo' => 'boolean',
+            'effective_start' => 'date',
+            'effective_end' => 'date',
+            'cost' => 'numeric'
+        ]);
+
+        $package = DuesPackage::find($id);
+        if ($package) {
+            $package->update($request->all());
+        } else {
+            return response()->json(['status' => 'error', 'message' => 'DuesPackage not found.'], 404);
```

Comment:
I want to push towards making the status methods consistent error codes in the style of the slack API. I've done it for the APIs I wrote, and we should go back and do the others.

This one could be `dues_package_not_found` or simply `entity_not_found`

Reply:
Let's discuss this further - not sure I see the benefit but if you want it I'm happy to change it.